pkg/kubectl/cmd/util/editor/editoptions.go

```diff
@@ -153,12 +157,34 @@ func (o *EditOptions) Run() error {
 			results  = editResults{}
 			original = []byte{}
 			edited   = []byte{}
-			file     string
 			err      error
 		)
 
 		containsError := false
 
+		if o.EditMode == ApplyEditMode {
```

Comment:
I'd rather see this done in the `switch o.EditMode {` below, so that the infos passed into the editFn are actually the ones we want to edit